cpp/ql/src/semmle/code/cpp/Variable.qll

```diff
@@ -126,8 +130,9 @@ class Variable extends Declaration, @variable {
     or
     exists(AssignExpr ae | ae.getLValue().(Access).getTarget() = this and result = ae.getRValue())
     or
-    exists(AggregateLiteral l |
-      this.getDeclaringType() = l.getType() and
+    exists(AggregateLiteral l, Class c |
+      this.(Field).getDeclaringType() = c and
+      aggregateLiteralHasType(l, c) and
       result = l.getChild(this.(Field).getInitializationOrder())
     )
```

Comment:
I think the modern way to express this whole case would be

```ql
exists(ClassAggregateLiteral l | result = l.getFieldExpr(this))
```

Does that perform better?

Reply:
Yes, it does perform better (it even avoids the join with all children). And it has the benefit of using `getUnspecifiedType` instead of `getType` so we get more results.